app/controllers/proposals/new.js

```diff
@@ -1,29 +1,24 @@
-import Ember from 'ember';
-import QueryParams from 'ember-parachute';
+import Controller from 'ember-controller';
+import { filterBy } from 'ember-computed';
+import injectService from 'ember-service/inject';
 
-export const queryParams = new QueryParams({
-  recipient: {
-    defaultValue: ''
-  },
-  amount: {
-    defaultValue: ''
-  },
-  url: {
-    defaultValue: ''
-  },
-  ipfsHash: {
-    defaultValue: ''
-  }
-});
-
-export default Ember.Controller.extend(queryParams.Mixin, {
+export default Controller.extend({
+  kredits: injectService(),
 
-  contributors: null,
+  contributors: [],
+  minedContributors: filterBy('contributors', 'id'),
 
   actions: {
-    onSave() {
-      this.transitionToRoute('index');
+    save(proposal) {
+      // contributorIpfsHash is needed for the proposal ipfs data. I'm not happy to do this here but I think to load all the contributors in addProposal again is a bit too much. I hope we can refactor it later.
```

Comment:
@bumi we need to look into hubot and how we should do the contributor look up there. I think if we have some kind of caching it would be better to have it in the kredits module

Reply:
I'd remove the contributor ipfsHash from the proposal and only have the kredits id stored there. With that one the latest profile ipfs hash can be retrieved from the contract. This would also make sure that one always gets the latest profile hash of the contributor. 

hope I did not forget some reason why the hash is stored there.

Reply:
It is stored there, because you need to be able to access the profile at the time of the proposal.

Reply:
hmmm. but all data is stored in the contract and that one holds the "truth". 
the proposal IPFS hash is retrieved from the contract and through the contracts the contributor ipfs hash can be retrieved. (the latest one, but also thanks to blockchain the one at a specific time in the past - when the proposal was created). 

To load data might be one more step: get the contributor by id from the contract and then access the contributor profile from IPFS. 

I think at least we should also add the kredits contributor id to the proposal ipfs data to be able to get to the contributor in the contract from the ipfs proposal data.